main/signers/lattice/Lattice/index.ts

```diff
@@ -257,15 +259,13 @@ export default class Lattice extends Signer {
     }
   }
 
-  async signTypedData (index: number, version: string, typedData: TypedData, cb: Callback<string>) {
-    const versionNum = (version.match(/[Vv](\d+)/) || [])[1]
-
-    if ((parseInt(versionNum) || 0) < 4) {
-      return cb(new Error(`Invalid version (${version}), Lattice only supports eth_signTypedData version 4+`))
+  async signTypedData (index: number, typedMessage: TypedMessage<SignTypedDataVersion.V4>, cb: Callback<string>) {
+    if (typedMessage.version !== SignTypedDataVersion.V4) {
```

Comment:
can this ever be the case with the given typings?

Reply:
Yeah this was on my TODO - I think the version check happens before it gets to the signers so I don't think we need a version check here